Python/compile.c

```diff
@@ -7731,7 +7731,13 @@ optimize_and_assemble_code_unit(struct compiler_unit *u, PyObject *const_cache,
     if (stackdepth < 0) {
         goto error;
     }
-    assert(stackdepth >= 0);
+
+    /* prepare_localsplus adds instructions for generators that push
+     * and pop an item on the stack. This assertion makes sure there
+     * is space on the stack for that.
+     * It should always be true, because at least one expression is
+     * required to turn a function into a generator.
```

Comment:
I'm not sure this explanation is accurate? This function is a generator:

```
async def f():
    return
```

Here is the disassembly of it:

```
>>> dis.dis(f)
  1           0 RETURN_GENERATOR
              2 POP_TOP
              4 RESUME                   0

  2           6 RETURN_CONST             0 (None)
        >>    8 CALL_INTRINSIC_1         3 (INTRINSIC_STOPITERATION_ERROR)
             10 RERAISE                  1
ExceptionTable:
  4 to 6 -> 8 [0] lasti
```

I think if `RETURN_CONST` didn't exist, it would be enough to say "all functions must return a value, which always must go on the stack before it is returned." But with `RETURN_CONST` that's not true either. It really does seem like the `StopIteration` handler is what we are relying on here.